compressor/management/commands/compress.py

```diff
@@ -210,46 +210,55 @@ def compress(self, engine, extensions, verbosity, follow_links, log):
                         nodes_count += 1
                         template_nodes.setdefault(node, []).append(context)
 
+            pool = concurrent.futures.ThreadPoolExecutor(max_workers=4)
```

Comment:
It might be better to use `ProcessPoolExecutor` so you don't need to worry about thread safety?

Otherwise, you might need to add locks around the places where it manipulates shared objects (like `offline_manifest`).

It's probably a good idea to make the number of workers configurable too?

Reply:
Actually nevermind, there might still be some potential bugs when manipulating the shared objects with `ProcessPoolExecutor`.

You might need to refactor this to avoid that?